src/gas_estimator.rs

```diff
@@ -105,10 +129,91 @@ impl GasTracker {
 
         let mut vector: Vec<&GasPriceEntry> = Vec::from_iter(self.history.iter());
         vector.sort();
-
         // this should never panic as percentage is less than 1 and vector len is
         // included as a factor
         let lowest: usize = (percentage * vector.len() as f32).floor() as usize;
         Some(vector[lowest].sample.clone())
     }
 }
+
```

Comment:
Here are two basic unit tests. One of them uses eth.althea.net to fetch a gas price, but the test might fail if that host is unreachable.